### PR TITLE
save json artifact

### PR DESCRIPTION
**Pull request by smark88**

### CODE REVIEW

File: .github/workflows/trackdeploys.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Track semVer through environments
+
+on:
+  schedule:
+    - cron: '0 */3 * * *' # run every 3 hours and on push
```
Comment (snf2ye) [inline .github/workflows/trackdeploys.yaml:5]: Why does this need to run both on push and on a schedule?

**Review by snf2ye — COMMENTED** [review]
Does this map to a ticket? I'd appreciate learning a bit more about the motivation behind this action!

**Comment by smark88** [comment]
> Does this map to a ticket? I'd appreciate learning a bit more about the motivation behind this action!

no ticket yet

File: .github/workflows/trackdeploys.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            else
+             echo "No env found"
+            fi
+            date=$(date +%Y-%m-%d\ %H:%M:%S)
+            semVer=$(curl -s $i | jq -r '.semVer')
+            cat <<< $(jq --arg env "$env" \
+              --arg date "$date" \
+              --arg semVer "$semVer" \
+              '.data += [{"env": "'"$env"'", "date": "'"$date"'", "semVer": "'"$semVer"'"}]' endpoints.json) \
+              > endpoints.json
```
Comment (fboulnois) [inline .github/workflows/trackdeploys.yaml:41]: I am not sure how well defined `cat <<< $(... $file) > $file` is.

**Review by fboulnois — APPROVED** [review]
A good first pass, one comment:

File: .github/workflows/trackdeploys.yaml (hunk 2 above)
Comment (smark88) [reply .github/workflows/trackdeploys.yaml:41]: unsure i stole it from here https://stackoverflow.com/questions/36565295/jq-to-replace-text-directly-on-file-like-sed-i

File: .github/workflows/trackdeploys.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_BUCKET_SA_KEY }}
+          export_default_credentials: true
+      - name: Use gcloud copy endpoints.json
+        run: gsutil cp gs://deployment-tracking/endpoints.json .
+      - shell: bash
+        run: |
+          endPoints+=( "https://jade.datarepo-dev.broadinstitute.org/configuration" \
```
Comment (pshapiro4broad) [inline .github/workflows/trackdeploys.yaml]: do you mean to append to `endPoints` here? I would expect to see
```suggestion
          endPoints=( "https://jade.datarepo-dev.broadinstitute.org/configuration" \
```

File: .github/workflows/trackdeploys.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            elif [[ "$i" == *"staging"* ]]
+            then
+             env=staging
+            elif [[ "$i" == *"data.terra.bio"* ]]
+            then
+             env=production
+            else
+             echo "No env found"
+            fi
+            date=$(date +%Y-%m-%d\ %H:%M:%S)
```
Comment (pshapiro4broad) [inline .github/workflows/trackdeploys.yaml]: This way, the dates for each release may be slightly different. Pulling `date` outside the `for` will give you the same date for all of them.

Comment (smark88) [reply .github/workflows/trackdeploys.yaml]: probably a good idae

Comment (smark88) [reply .github/workflows/trackdeploys.yaml]: date moved out of for loop

File: .github/workflows/trackdeploys.yaml (hunk 3 above)
Comment (smark88) [reply .github/workflows/trackdeploys.yaml]: committed suggestion

File: .github/workflows/trackdeploys.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      - shell: bash
+        run: |
+          endPoints+=( "https://jade.datarepo-dev.broadinstitute.org/configuration" \
+            "https://data.alpha.envs-terra.bio/configuration" \
+            "https://data.staging.envs-terra.bio/configuration" \
+            "https://data.terra.bio/configuration")
+          date=$(date +%Y-%m-%d\ %H:%M:%S)
+
+          for i in ${endPoints[@]}; do
+            if [[ "$i" == *"dev"* ]]
```
Comment (nmalfroy) [inline .github/workflows/trackdeploys.yaml]: Ah man I wish we could use objects here...

Comment (nmalfroy) [reply .github/workflows/trackdeploys.yaml]: Actually, could this just be in an endpoints.json so we didn't have to do bash logic?

**Review by nmalfroy — COMMENTED** [review]
Quick question about maybe making this easier if we just used json to describe the environments up front

File: .github/workflows/trackdeploys.yaml (hunk 5 above)
Comment (smark88) [reply .github/workflows/trackdeploys.yaml]: Switched to  jq to ditch those ifs

File: .github/workflows/trackdeploys.yaml (hunk 2 above)
Comment (nmalfroy) [reply .github/workflows/trackdeploys.yaml:41]: Looks like it's literally the string to the right

Comment (nmalfroy) [reply .github/workflows/trackdeploys.yaml:41]: Oh, you meant like order of operations?

Comment (nmalfroy) [reply .github/workflows/trackdeploys.yaml:41]: just checking: this prints the contents of endpoints.json to the console?  Be good if it did for tracking in the logs

Comment (smark88) [reply .github/workflows/trackdeploys.yaml:41]: it doesn't since its redirected to the file

**Review by snf2ye — APPROVED** [review]
Nice - https://storage.cloud.google.com/deployment-tracking/endpoints.json - Looks good!

File: .github/workflows/trackdeploys.yaml (hunk 2 above)
Comment (fboulnois) [reply .github/workflows/trackdeploys.yaml:41]: Yes, I meant order of operations. It redirects the output of the jq command that reads in a file and outputs it to that same file. It looks like it works okay though.

**Review by fboulnois — APPROVED** [review]

**Pull request merged**